zerver/lib/webhooks/common.py

```diff
@@ -114,3 +116,32 @@ def validate_extract_webhook_http_header(request: HttpRequest, header: str,
         raise MissingHTTPEventHeader(header)
 
     return extracted_header
+
+class AuthStrategies:
+    SIMPLE_TOKEN = 1
+
+def http_api_callback(bot: UserProfile, uri: str)-> Dict[str, Any]:
+    try:
+        api_callback_details = json.loads(BotConfigData.objects.get(key="api_callback_details").value)
+        auth_strategy = api_callback_details["auth_strategy"]
+    except BotConfigData.DoesNotExist:
+        raise Exception  # TODO: Refine how we notfiy users to configure the auth_details.
+    except KeyError:
+        raise Exception  # TODO
```

Comment:
We'll probably want some better exceptions, perhaps subclass `JsonableError`?

Reply:
Definitely. I was planning on doing that after I thought of what good error messages would be :)